CommunityToolkit.Net.Authentication/IProvider.cs

```diff
@@ -3,13 +3,10 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.ComponentModel;
-using System.Net.Http;
-using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 using Microsoft.Graph;
```

Comment:
The `IAuthenticationProvider` is the sticking point for our reliance on the Graph here, eh?

Do we maybe not add this explicit interface here and then do that at the higher-level for a Graph Provider? So like:

IProvider
BaseProvider : IProvider
----
MsalProvider : IProvider
----
WindowsProvider : IProvider
----
GraphProvider : IAuthenticationProvider (and can take an Msal or Window one in somehow?)

Reply:
Reliance relieved! We now use the `DelegateAuthenticationProvider` and don't require a reference to the `IAuthenticationProvider` interface.